tests/drivers/gpio/egpio_basic_api/dts/bindings/test-egpio.yaml

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+description: |
+    This binding provides resources required to build and run the
+    tests/drivers/gpio/egpio_basic_api test in Zephyr.
+
+compatible: "test-egpio"
+
+properties:
+  out-gpios:
+    type: phandle-array
+    required: true
+    description: |
+      Identity of a eGPIO that will be configured as an output.
+      This must be on the instance to which FLPR has access, and physically
+      connected to in-gpios.
+
+  in-gpios:
+    type: phandle-array
+    required: true
+    description: |
+      Identity of a hardware GPIO that will be configured as an input.
+      This must be on the same instance as out-gpios, and physically
```

Comment:
I would say input gpio does not have to be on the same port as output gpio, I can see there are different gpio ports used in the overlay file. I would even say they SHOULDN'T use the same port because there might be a conflict in handling the port event.